docs/hoon/library/1b.md

```diff
@@ -1,147 +1,294 @@
-            ?(%0 %1)  !!
-            *         $(a (div a 2))
-          ==
-        ::
+  `a` is an atom.
 
-Examples
---------
+  Produces
+  --------
 
-    > (cap 4)
-    %2
-    > (cap 6)
-    %3
-    > (cap (add 10 9))
-    %2
+  A constant atom.
 
+  Source
+  ------
 
+          ++  cap                                                 ::  tree head
+            ~/  %cap
+            |=  a=@
+            ^-  ?($2 $3)
+            ?-  a
+              $2        %2
+              $3        %3
+              ?(%0 %1)  !!
+              *         $(a (div a 2))
+            ==
+          ::
 
-***
-### `++mas`
+  Examples
+  --------
 
-Axis within head/tail?
+      > (cap 4)
+      %2
 
-Computes the axis of `a` within either the head or tail of a noun (depends whether `a` lies within the the head or tail).
+      > (cap 6)
+      %3
 
-Accepts
--------
+      > (cap (add 10 9))
+      %2
 
-`a` is an [atom]().
+      > (cap 1)                                    ::axis '0' is in neither the head nor the tail
+      ! exit
 
-Produces
---------
+      > (cap 0x40))                                ::0x1a is of the hexadecimal soft atom type (@ux)
+      %2
+      > `@`0x40
+      64                           
 
-An atom.
+      > (cap 'a')                                  ::'a' is of the cord (text) soft atom type (@t)
+      %3
+      > `@`'a'
+      97                                        
 
-Source
-------
 
-        ++  mas                                                 ::  tree body
-          ~/  %mas
-          |=  a=@
-          ^-  @
-          ?-  a
-            1   !!
-            2   1
-            3   1
-            *   (add (mod a 2) (mul $(a (div a 2)) 2))
-          ==
-        ::
+  ***
+  ### `++mas`
 
-Examples
---------
+  Address within head/tail?
 
-    > (mas 3)
-    1
-    > (mas 4)
-    2
-    > (mas 5)
-    3
-    > (mas 6)
-    2
-    > (mas 0)
-    ! exit
-    > (mas 1)
-    ! exit
+  Computes the tree address of atom `a` within either the head (`+2`) or tail
+  (`+3`) of a noun. In other words, `++mas` returns the relative address of `a`
+  with respect to the head or the tail subtrees, depending on which one `a` 
+  descends from. 
+  
+              1
+            /   \
+           /     \
+          2       3              <--here are the head (`+2`) and the tail (`+3`)     
+         / \      /\
+        4   5    6  7
+       /\   /\  /\  /\
+       (continues...)
 
+  Running `(mas 7)` in the Dojo will return `3`, because address `+3` is what 
+  `+7` now occupies. The tree below helps illustrate the relationship. With 
+  parentheses are `a` values (if `a` is in subtree `+3`), and without parentheses 
+  are the values returned with `(mas a)`.
 
+              1(3)
+             /    \
+            2       3
+           (6)     (7)
+          / \       /\
+         /   \     /  \
+        4     5   6    7
+       (12) (13) (14) (15)
+      / \    / \ / \   / \
+         (continues...)
```

Comment:
This explanation is good.

Reply:
Maybe you should have a comment in the diagram that's something like: `::  new/(old) addresses`.